### PR TITLE
add api key

### PR DESCRIPTION
**Pull request by undeaDD**

### CODE REVIEW

**Comment by undeaDD** [comment]
The code changes you provided seem to add an API key to a URL in the getPokemonBy(nameOrId:) function in the ApiManager.swift file. Let's analyze these changes:

---

API Key Addition:

```
+ // add static api key for authentication
+ let API_KEY = "fa877ee9-a9ac-4b59-b725-601025b1e946"
```
This part of the change is adding a static API key to the code, which is often used for authentication purposes. Adding an API key this way is generally fine as long as it's not sensitive information, and it's kept safe.

URL Construction:

```
- guard let url = URL(string: Utils.baseURL + nameOrId) else {
+ guard let url = URL(string: Utils.baseURL + nameOrId + API_KEY) else {
```
The change modifies the construction of the URL by appending the API key to the end of the URL. This seems to be the intended change based on the comment above.

---

However, there are a few things to consider:

**API Key Security**: Ensure that the API key is not sensitive, or if it is sensitive, make sure it's properly protected. Hardcoding API keys in the source code is generally discouraged for sensitive keys because they can be easily exposed if the code is publicly available.

**Separation of Concerns**: It's often a good practice to keep API keys separate from source code. Using environment variables or a configuration file to store API keys can make it easier to manage and protect them.

**Error Handling**: The throws keyword in the function signature suggests that this function can throw an error of type ApiError.invalidName. Make sure that this error is handled correctly in the calling code, or if this is an expected error, handle it appropriately within this function.

**Constants**: If the API key is intended to be a constant value, consider declaring it as a static constant property within the ApiManager class or in a separate configuration file for better organization.

In summary, the code changes appear to serve the purpose of adding an API key to the URL for authentication. However, you should take additional steps to secure the API key, consider separating it from the source code, and handle errors appropriately.
